### PR TITLE
Update build.sbt for publishing.

### PR DESCRIPTION
**Pull request by grebe**
This should help people use rocket as an ivy dependency. Duplicates functionality from #1319.

@ucbjrl, would appreciate your feedback.

### CODE REVIEW

**Comment by edcote** [comment]
My "manager" feedback is to ask why are you doing more than #1319?

Can you justify changing organization/version versions?  This could break someone else's API.  Imho, this would be at the discretion of the project's maintainers, not lowly contributors such as ourselves.

What value do you get from aggregate more than rocketchip level?

**Comment by grebe** [comment]
This is to make it possible to publish on a repo like sonatype so you can use rocket-chip as an ivy dep. @ucbjrl publishes artifacts for edu.berkeley.cs. I'm setting the version to a snapshot, which is the standard practice for ivy stuff when it's not a release.

As far as I know, nobody who is a maintainer has any plans to use rocket as an ivy dep, so this change should be irrelevant to them, but I know several people who would like to use components from rocket in other projects (e.g. just to grab an async Fifi).

Maybe aggregating the subjects is unnecessary. Sometimes I'll mess around with a subproject and publish local the subproject rather than the whole thing to save time, so it's nice to aggregate them properly, but that's pretty niche.

File: build.sbt [hunk 1]
```diff
@@ -20,11 +20,17 @@ lazy val commonSettings = Seq(
 )
 
 lazy val chisel = (project in file("chisel3")).settings(commonSettings)
-lazy val hardfloat  = project.dependsOn(chisel).settings(commonSettings)
-lazy val macros = (project in file("macros")).settings(commonSettings)
+lazy val hardfloat  = project.dependsOn(chisel).
+                              aggregate(chisel).
+                              settings(commonSettings)
```
Comment (hcook) [inline build.sbt]: can you format these with the same order and spacing as `rocketchip` below?

File: build.sbt [hunk 2]
```diff
@@ -20,11 +20,17 @@ lazy val commonSettings = Seq(
...
+lazy val macros = Project(
+        id = "rocket-macros",
+        base = file("macros")).
+        settings(commonSettings)
```
Comment (hcook) [inline build.sbt]: Can you left align these two spaces?

**Comment by edcote** [comment]
@hcook ack.

**Comment by grebe** [comment]
Bump

**Comment by grebe** [comment]
Closed by #1835

**Pull request closed without merging**